frontend/src/components/sustainability/SustainabilityBoard.js

```diff
@@ -44,22 +48,34 @@ class SustainabilityBoard extends Component {
                 // Cell's board not loaded.
             }
         } else if (range === 'user_board') {
-            data.billable = accounts.billable_accounts.by_person[id];
-            data.non_billable = accounts.non_billable_accounts.by_person[id];
-            data.non_billable_responsible = accounts.non_billable_responsible_accounts.by_person[id];
+            data.billable = accounts.billable_accounts.by_person[id] || 0;
+            data.non_billable = accounts.non_billable_accounts.by_person[id] || 0;
+            data.non_billable_responsible = accounts.non_billable_responsible_accounts.by_person[id] || 0;
         } else {
-            data.billable = accounts.billable_accounts.overall;
-            data.non_billable = accounts.non_billable_accounts.overall;
-            data.non_billable_responsible = accounts.non_billable_responsible_accounts.overall;
+            data.billable = accounts.billable_accounts.overall || 0;
+            data.non_billable = accounts.non_billable_accounts.overall || 0;
+            data.non_billable_responsible = accounts.non_billable_responsible_accounts.overall || 0;
         }
         data.non_billable_total = data.non_billable + data.non_billable_responsible;
-        data.responsible_ratio = data.non_billable_responsible / data.billable * 100;
-        data.total_ratio = data.non_billable_total / data.billable * 100;
-        data.remaining = data.billable * MAX_NON_BILLABLE_TO_BILLABLE_RATIO - data.non_billable_responsible;
+        data.total = data.billable + data.non_billable_total;
+        data.total_ratio = data.non_billable_total / data.total * 100;
+        data.remaining = data.billable * MAX_NON_BILLABLE_TO_BILLABLE_RATIO / (1 - MAX_NON_BILLABLE_TO_BILLABLE_RATIO) - data.non_billable_total;
+        data.cell_hours = data.billable + data.non_billable_responsible;
+        data.responsible_ratio = data.non_billable_responsible / data.cell_hours * 100;
+        data.remaining_responsible = data.billable * MAX_NON_BILLABLE_TO_BILLABLE_CELL_RATIO / (1 - MAX_NON_BILLABLE_TO_BILLABLE_CELL_RATIO) - data.non_billable_responsible;
```

Comment:
I have many comments and questions about these calculations. [I posted in the forum](https://forum.opencraft.com/t/sustainability-dashboard/431/29). I don't think we have time to wait for the forum responses in order to finish this task.
I also don't have time to do a full review and to try all cases and improve all names and explanations. If you have tested this and don't see anything wrong after the forum comments, we can merge it.
(It's hard for me to say this. Usually when I approve a review, I'm certain about the correctness of the approach. Here, we didn't have time).

I read the formulas and they seem to make sense, though I didn't debug them. I also arrived to the same formula for the last one: ` nb/(b+nb)=0'3 → nb=0'3b+ 0'3nb → nb = b * 0'3 / (1-03).` (and then deducting the already spent nb).

Reply:
I derived these formulas basing on our assumptions for this change, but didn't spend time on making mathematical proofs for them, just checked them for a few cases.